crates/aptos/src/common/types.rs

```diff
@@ -349,16 +368,25 @@ pub struct WriteTransactionOptions {
     pub private_key_options: PrivateKeyInputOptions,
     #[clap(flatten)]
     pub rest_options: RestOptions,
-    /// ChainId for the network
-    #[clap(long)]
-    pub chain_id: ChainId,
     /// Maximum gas to be used to publish the package
     ///
     /// Defaults to 1000 gas units
     #[clap(long, default_value_t = 1000)]
     pub max_gas: u64,
 }
 
+impl WriteTransactionOptions {
+    pub async fn chain_id(&self) -> CliTypedResult<ChainId> {
+        let client = Client::new(self.rest_options.url()?);
+        let state = client
+            .get_ledger_information()
+            .await
+            .map_err(|err| CliError::ApiError(err.to_string()))?
+            .into_inner();
+        Ok(ChainId::new(state.chain_id))
```

Comment:
can we cache this so we don't make a request every time?

Reply:
Cut an issue for it, it's not urgent